pcsx2-qt/Debugger/MemorySearchWidget.cpp

```diff
@@ -123,9 +118,22 @@ void MemorySearchWidget::onListSearchResultsContextMenu(QPoint pos)
 	contextMenu->popup(m_ui.listSearchResults->viewport()->mapToGlobal(pos));
 }
 
+template<typename T>
+T readValueAtAddress(DebugInterface* cpu, u32 addr);
+template<>
+float readValueAtAddress<float>(DebugInterface* cpu, u32 addr)
+{
+	return std::bit_cast<float>(cpu->read32(addr));
+}
+
+template<>
+double readValueAtAddress<double>(DebugInterface* cpu, u32 addr)
+{
+	return std::bit_cast<double>(cpu->read64(addr));
+}
```

Comment:
I'm thinking instead of having these functions for `float` and `double`, you can instead bitcast _every_ result in the generic `T` function below

Such as
```cpp
template <typename T>
T readValueAtAddress(DebugInterface* cpu, u32 addr)
{
	T val = 0;
	switch (sizeof(T))
	{
		case sizeof(u8):
			val = std::bit_cast<T>(cpu->read8(addr));
			break;
		case sizeof(u16):
			val = std::bit_cast<T>(cpu->read16(addr));
			break;
```

(This is free and wont have a performance hit, if you're worried about that)

Reply:
Trying to incorporate this feedback, I get 'std::bit_cast': no matching overloaded function found` errors. I haven't found a solution for that yet. I like this idea, however for now I will leave this as it is until I find a way to make it work.